### PR TITLE
feat: obtain an ordinal number of type String for any integer

### PR DESCRIPTION
**Pull request by Biplab-Dutta**

### CODE REVIEW

File: lib/src/int.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+extension IntSuffix<T extends int> on T {
+  /// Returns a string adding a proper suffix on any integer
+  ///
+  /// ```dart
+  /// 101.addSuffix(); // 101st
+  ///
+  /// 999218.addSuffix(); // 999218th
+  /// ```
+  String addSuffix() {
```
Comment (passsy) [inline lib/src/int.dart:11]: I'd call it `ordinal()`

**Comment by passsy** [comment]
Interesting. Neither the [`intl`](https://github.com/dart-lang/intl/issues/226) nor the [`flutter`](https://github.com/flutter/flutter/issues/37221) maintainers feel responsible for this. I personally would add it to `intl`.

**Comment by passsy** [comment]
Try it, but I honestly don't have high hopes of it getting merged there. I'm happy to integrate it into `dartx`

File: lib/src/int.dart (hunk 1 above)
Comment (Biplab-Dutta) [reply lib/src/int.dart:11]: Aaahh Thanks. I was exactly looking for this very word 😂
yes ordinal is good one
Thanks

**Comment by Biplab-Dutta** [comment]
I was reading some old conversations there. I am not sure of the outcome. Is it okay if I send them a PR and see what they have to say and if they don't like it, I'd then send you another PR with some updates in the doc too..

**Comment by passsy** [comment]
Sure :)

**Comment by Biplab-Dutta** [comment]
By the way, I am not that familiar with workflows. I am seeing some unsuccessful checks. Does that have anything to do with my code or am I supposed to do something else to have all 8 checks successful?

**Comment by Biplab-Dutta** [comment]
I think it is now good to be merged.

File: lib/src/int.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  /// ```dart
+  /// 101.ordinal(); // 101st
+  ///
+  /// 999218.ordinal(); // 999218th
+  /// ```
+  String ordinal() {
+    final onesPlace = this % 10;
+    final tensPlace = ((this / 10).floor()) % 10;
+    if (tensPlace == 1) {
+      return '${this}th';
```
Comment (passsy) [inline lib/src/int.dart:15]: Please add tests for 10-15 :)

Comment (Biplab-Dutta) [reply lib/src/int.dart:15]: All right.. looks like I missed few 😅 Shall be sending another PR in a moment

Comment (passsy) [reply lib/src/int.dart:15]: Especially 101st should still be part of the tests, that was great

Comment (Biplab-Dutta) [reply lib/src/int.dart:15]: Done!!

**Comment by passsy** [comment]
Thanks for your contribution!

**Comment by Biplab-Dutta** [comment]
> Thanks for your contribution!

My pleasure.

**Comment by oravecz** [comment]
Is this en locale only? Is ordinal a concept in other locales and are they handled differently?

**Comment by passsy** [comment]
This method here is 'en' only. It does exist in other languages, see [Ordinal indicator](https://en.wikipedia.org/wiki/Ordinal_indicator).

We could add a `locale` parameter to support more locales

**Pull request merged**